File: backend/DAL/Entities/ExpertRequest.cs
using DAL.Entities;

namespace DAL;

public class ExpertRequest : Entity {
  public Guid UserId { get; set; }
  public string Description { get; set; }
  public DateTime? ApprovedAt { get; set; }
  public Guid? ApprovedUserId { get; set; }

  // Foreign Keys
  public User User { get; set; }
  public User? ApprovedUser { get; set; }
}

File: frontend/src/services/entities/ExpertRequest.ts
import { User } from "./User";


export interface ExpertRequest {
    id?: string;
    createdAt?: Date;
    updatedAt?: Date | undefined;
    userId?: string;
    description?: string | undefined;
    approvedAt?: Date | undefined;
    approvedUserId?: string | undefined;
    user?: User;
    approvedUser?: User;
}
